src/overlays/actors/ovl_Demo_Du/z_demo_du.c

```diff
@@ -1,228 +1,1083 @@
+    }
+}
+
+// Cutscene: Darunia gives Link the Goron's Ruby.
+// Sfx played when Link is surprised by Darunia falling from the sky.
+void DemoDu_CsPlaySfx_LinkSurprised(GlobalContext* globalCtx) {
+    Player* player;
+
+    if (globalCtx->csCtx.frames == 174) {
+        player = PLAYER;
+        Audio_PlaySoundGeneral(NA_SE_VO_LI_SURPRISE_KID, &player->actor.projectedPos, 4U, &D_801333E0, &D_801333E0,
+                               &D_801333E8);
+    }
+}
+
+void DemoDu_CsGoronsRuby_UpdateFaceTextures(DemoDu* this, GlobalContext* globalCtx) {
+    u16* frames = &globalCtx->csCtx.frames;
+
+    if (*frames < 260) {
+        DemoDu_UpdateEyes(this);
+        DemoDu_SetMouthTexIndex(this, 0);
+    } else if (*frames < 335) {
+        DemoDu_UpdateEyes(this);
+        DemoDu_SetMouthTexIndex(this, 3);
+        return;
+    } else if (*frames < 365) {
+        DemoDu_SetEyeTexIndex(this, 3);
+        DemoDu_SetMouthTexIndex(this, 1);
+        return;
+    } else if (*frames < 395) {
+        DemoDu_SetEyeTexIndex(this, 0);
+        DemoDu_SetMouthTexIndex(this, 3);
+        return;
+    } else if (*frames < 410) {
+        DemoDu_UpdateEyes(this);
+        DemoDu_SetMouthTexIndex(this, 0);
+        return;
+    } else {
+        DemoDu_UpdateEyes(this);
+        DemoDu_SetMouthTexIndex(this, 3);
+    }
+}
+
+void func_8096A630(DemoDu* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Vec3f pos = this->actor.world.pos;
+
+    pos.y += kREG(5);
+    func_80033480(globalCtx, &pos, kREG(1) + 100.0f, kREG(2) + 0xA, kREG(3) + 0x12C, kREG(4), 0);
+    DemoDu_CsPlaySfx_GoronLanding(this);
+}
+
+static Vec3f sDustVelocity = { 0.0f, 0.0f, 0.0f };
+static Vec3f sDustAccel = { 0.0f, 0.3f, 0.0f };
+static Color_RGBA8 sDustPrimaryColor = { 190, 150, 110, 255 };
+static Color_RGBA8 sDustEnvironmentColor = { 120, 80, 40, 255 };
+static Vec3f sDustPosOffsets[] = { { 11.0f, -11.0f, -6.0f }, { 0.0f, 14.0f, -13.0f }, { 14.0f, -2.0f, -10.0f },
+                                   { 10.0f, -6.0f, -8.0f },  { 8.0f, 6.0f, 8.0f },    { 13.0f, 8.0f, -10.0f },
+                                   { -14.0f, 1.0f, -14.0f }, { 5.0f, 12.0f, -9.0f },  { 11.0f, 6.0f, -7.0f },
+                                   { 14.0f, 14.0f, -14.0f } };
+
+void DemoDu_CsGoronsRuby_SpawnDustWhenHittingLink(DemoDu* this, GlobalContext* globalCtx) {
+    Player* player;
+    s32 i;
+    s32 pad[2];
+    Vec3f* headPos;
+    Vec3f velocity;
+    Vec3f accel;
+    s32 colorDelta;
+    Color_RGBA8 primColor;
+    Color_RGBA8 envColor;
+    SkelAnime* skelAnime = &this->skelAnime;
+    Vec3f position;
+
+    if (Animation_OnFrame(skelAnime, 31.0f) || Animation_OnFrame(skelAnime, 41.0f)) {
+        player = PLAYER;
+        headPos = &player->bodyPartsPos[PLAYER_LIMB_HEAD];
+        velocity = sDustVelocity;
+        accel = sDustAccel;
+
+        for (i = 4; i >= 0; --i) {
+            primColor = sDustPrimaryColor;
+            envColor = sDustEnvironmentColor;
```

Comment:
These are probably inlined assignments. You may need to fiddle with the stack to make them work out.